### PR TITLE
Validate / fix user-storage-jpa

### PR DESCRIPTION
**Pull request by vramik**
There has been two separate quickstarts.

- `user-storage-jpa-legacy` was made for legacy wildfly dist and contained an integration test.
- `user-storage-jpa` was modified for quarkus dist, but it didn't have any integration test.

The PR merges two quickstarts together. 

- `user-storage-jpa-legacy` was removed
- integration test was moved into `user-storage-jpa`
- `user-storage-jpa` was enabled for GHA
- `user-storage-jpa` now implements non-deprecated methods

Closes keycloak/keycloak#17392
Closes keycloak/keycloak#17391

### CODE REVIEW

File: user-storage-jpa/src/test/resources/MANIFEST.MF
Comment (ahus1) [inline user-storage-jpa/src/test/resources/MANIFEST.MF:1]: Could this file be removed? I don't think it is used by the Quarkus dist.

**Review by ahus1 — CHANGES_REQUESTED** [review]
Hi @vramik - thank you for making this work with tests for the Quarkus distribution!

See below for some comment. I didn't find the time to start it manually today; I plan to do this for the next review iteration. 

Please see below for some comments.

File: user-storage-jpa/README.md [hunk 1]
```diff
@@ -50,6 +50,18 @@ Add the provider, save it.  This will now enable the provider for the 'master' r
 admin console or on the registration pages of <span>Keycloak</span>, will be created in the custom store used by the provider.  If you go
 to the Users tab in the Admin Console and create a new user, you'll be able to see the provider in action.
 
+Integration test of the Quickstart
+----------------------------------
+
+1. Make sure you have an Keycloak server running with installed provider and with an admin user in the `master` realm. Your <span>Keycloak</span> should be listening on `http://localhost:8180/auth`. You can archive this by running:
```
[the remark this answers is not in the document]
Comment (vramik) [reply user-storage-jpa/README.md]: > Could we make this run in a more standard way without the` /auth`? The only other project that would need to change would be the startup and the `user-storage-simple`, which we "own" as well AFAIK + the startup of Keycloak for the tests.

Good point, unfortunately it's as straightforward as it seems. QS tests uses classes (`FluentTestsHelper` and `TestsHelper`) from main keycloak repo which are shared with testsuite there. There may be a way to configure different base url, it needs to be investigated.

Considering it, I'd suggest to tackle it in separate issue.

[the remark this answers is not in the document]
Comment (vramik) [reply user-storage-jpa/README.md]: https://github.com/keycloak/keycloak-quickstarts/issues/389

File: user-storage-jpa/src/test/resources/MANIFEST.MF
Comment (ahus1) [reply user-storage-jpa/src/test/resources/MANIFEST.MF:1]: Thanks, resolved.

File: user-storage-jpa/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ahus1) [reply user-storage-jpa/README.md]: Thanks, resolved.

**Review by ahus1 — APPROVED** [review]
Thank you for making these quick starts work again!

I've now also tested this manually, they work, and it is ready to be merged.

**Review by mhajas — APPROVED** [review]
Approving per @ahus1 's approval

**Pull request merged**